python/setuputils.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (c) 2018, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import re
+import subprocess
+import warnings
+
+
+def clone_repo(name, GIT_REPOSITORY, GIT_TAG, force_clone=False):
+    """
+    Function to clone repos if they have not been cloned already.
+    Variables are named identical to the cmake counterparts for clarity,
+    in spite of not being very pythonic.
+
+    Parameters
+        ----------
+        name : String
+            Name of the repo to be cloned
+        GIT_REPOSITORY : String
+            URL of the repo to be cloned
+        GIT_TAG : String
+            commit hash or git hash to be cloned. Accepts anything that
+            `git checkout` accepts
+        force_clone : Boolean
+            Set to True to ignore already cloned repositories in
+            external_repositories and clone
+
+    """
+
+    if not os.path.exists("external_repositories/" + name) or force_clone:
+        print("Cloning repository "
+              + name
+              + " into external_repositories/"
+              + name)
+        subprocess.check_call(['git', 'clone',
+                               GIT_REPOSITORY,
+                               'external_repositories/' + name])
+        wd = os.getcwd()
+        os.chdir("external_repositories/" + name)
+        subprocess.check_call(['git', 'checkout',
+                              GIT_TAG])
+        os.chdir(wd)
+    else:
+        print("Found repository "
+              + name
+              + " in external_repositories/"
+              + name)
+
+
+def get_repo_cmake_info(names, file_path):
+    """
+    Function to find information about submodules from cpp/CMakeLists file
```

Comment:
Update comment to something like: "Returns the git tags from each of the repos in names"? The return was not clear from the comment.